### PR TITLE
Login register and model changes

### PR DESCRIPTION
**Pull request by hvoliveira**
NEW - User login and register mechanism
Also, changed models... not all endpoints are functional after these changes (not in business logic sense, at least)
NEW - Swagger documentation! This feature is in its infancy, but is already a huge plus. Will be regularly updated.

Once again, devs will need to run requirements again after this PR is merged.

### CODE REVIEW

File: server/appbackend/serializers.py [hunk 1]
```diff
@@ -46,6 +48,7 @@ class ClassGroupSerializer(serializers.ModelSerializer):
     teacher = serializers.ReadOnlyField(source='teacher.first_name')
     collaborators = CollaboratorSerializer(many=True, read_only=True)
     course = serializers.ReadOnlyField(source='course.name')
+    semester = serializers.CharField(source='semester.name')
```
Comment (xFayre) [inline server/appbackend/serializers.py]: Semester is an Integer, perhaps deserializing it as charfield is causing the error

Comment (hvoliveira) [reply server/appbackend/serializers.py]: It isn't.

Comment (hvoliveira) [reply server/appbackend/serializers.py]: By the way, the problem is in Enrollment, this is ClassGroup

File: server/appbackend/serializers.py [hunk 2]
```diff
@@ -3,27 +3,35 @@
...
+        fields = '__all__'
+
+
 class StudentSerializer(serializers.ModelSerializer):
-    role = serializers.CharField(source='role.name')
 
     class Meta:
         model = Person
-        fields = ('id', 'name', 'phone', 'role')
+        fields = ('id', 'first_name', 'last_name', 'phone')
```
Comment (xFayre) [inline server/appbackend/serializers.py]: Why change from name to "first name and last name"?

Comment (hvoliveira) [reply server/appbackend/serializers.py]: Default django User fields, already are implemented. If the front end prefers to use only name and split it before sending to back end, we can do that.

File: server/appbackend/models.py [hunk 3]
```diff
@@ -1,52 +1,65 @@
...
+        PENDING = 'Pendente'
+        ACCEPTED = 'Aceita'
+        REJECTED = 'Rejeitada'
 
 
-class TransferStatus(Enum):
-    PENDING = 'p'
-    ACCEPTED = 'a'
+class EnrollmentStatus(Enum):
+    PENDING = 'p',
```
[the remark this answers is not in the document]
Comment (hvoliveira) [reply server/appbackend/models.py]: commited wrong fucking file

**Pull request merged**